record_web/lib/record_web.dart

```diff
@@ -145,15 +159,34 @@ class RecordPluginWeb extends RecordPlatform {
     return devices;
   }
 
+  double getMaxAmplitude() {
+    assert(analyser != null);
+    analyser?.getFloatFrequencyData(amplitudeDataArray!);
+    var maxAmplitude = amplitudeDataArray!.reduce(max);
+    return maxAmplitude;
+  }
+
   void _onStart(html.MediaStream stream) {
     _mediaRecorder = html.MediaRecorder(stream);
     _mediaRecorder?.addEventListener('dataavailable', _onDataAvailable);
     _mediaRecorder?.addEventListener('stop', _onStop);
     _mediaRecorder?.start();
-
+    createAudioContext(stream);
     _updateState(RecordState.record);
   }
 
+  createAudioContext(html.MediaStream stream) {
+    audioCtx = AudioContext();
+    var source = audioCtx!.createMediaStreamSource(stream);
+    analyser = audioCtx!.createAnalyser();
+    analyser?.minDecibels = -90;
+    analyser?.maxDecibels = -10;
+    source.connect(analyser!);
+    analyser?.fftSize = 256;
+    var bufferLength = analyser!.frequencyBinCount;
+    amplitudeDataArray = Float32List(bufferLength.toInt());
```

Comment:
`amplitudeDataArray` is never refreshed?
Maybe the name of the variable is misleading me...

Reply:
on javascript side the Float array is filled with data each call to getFloatFrequencyData on analyser